Models/Pool.sol

```diff
@@ -85,10 +89,10 @@ contract Pool {
     } // onlyWhenThePoolHasConculded()
 
     //----------------------------------
-    // Funcions
+    // Functions
     //----------------------------------
-    constructor(uint _ownerid) {
-      id = getNextUniqueId();
+    constructor(address _ownerid, uint _uniqueId, bool participateInThisPool) {
```

Comment:
Constructor should accept pool asset, stake asset, their respective amounts, cycle interval and maximum number of members as a parameter rather than hardcoding them.